app/address/[address]/concurrent-merkle-tree/page-client.tsx

```diff
@@ -0,0 +1,27 @@
+'use client';
```

Comment:
You don't actually need to split these into two files and `use client` unless there are _actual_ client functions in here (like `useCluster()`). Can you un-do this, and collapse it back into one file called `page.tsx` without the `use client`? The goal is to push `use client` as far out to the leaves as possible.

Reply:
Wait, doesn't the `ParsedAccountRender` use `useCluster` when fetching accountData? So I think this split should be okay, right?

Reply:
Oof. Hunh. What a mess.